src/debug/telemetry/telemetry2_thread_info.c

```diff
@@ -0,0 +1,349 @@
+/* SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) 2024 Intel Corporation
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <stdio.h>
+#include <soc.h>
+#include <adsp_debug_window.h>
+
+#include "telemetry2_slot.h"
+
+LOG_MODULE_REGISTER(thread_info);
+
+/* to adsp backend header, data to be shared with user space client*/
+#define THREAD_INFO_MAX_THREADS 16
+
+struct thread_info {
+	char name[14];
+	uint8_t stack_usage;	/* precentage */
+	uint8_t cpu_usage;	/* precentage */
+} __packed;
+
+#define THREAD_INFO_VERSION_0_0		0
+
+#define THREAD_INFO_STATE_UNINITIALIZED	0
+#define THREAD_INFO_STATE_BEING_UPDATED	1
+#define THREAD_INFO_STATE_UPTODATE	2
+
+struct thread_info_core {
+	uint8_t state;
+	uint8_t counter;		/* incremented every round */
+	uint8_t load;			/* precentage */
```

Comment:
percentage 

 but of what btw? per cpu? for total system?

Reply:
Per cpu... hmm should change that to fixed point value too. I'll add more specific comment.